### PR TITLE
chore: first iteration on refactoring WalletOverview

### PR DESCRIPTION
**Pull request by nicolasburtey**

### CODE REVIEW

File: app/components/text-currency/text-currency.tsx [hunk 1]
```diff
@@ -31,17 +31,17 @@ export const TextCurrencyForAmount: React.FC<Props> = ({
   iconColor,
 }) => {
   const { formatToDisplayCurrency } = useDisplayCurrency()
```
Comment (UncleSamtoshi) [inline app/components/text-currency/text-currency.tsx:33]: Is this component still used anywhere?

File: app/screens/send-bitcoin-screen/payment-details/onchain.ts [hunk 2]
```diff
@@ -22,6 +22,8 @@ export const createNoAmountOnchainPaymentDetails = <T extends WalletCurrency>(
   params: CreateNoAmountOnchainPaymentDetailsParams<T>,
 ): PaymentDetail<T> => {
   const {
+    // FIXME: is that coming from usePriceConversion()
+    // in that case we should not pass it as a param
```
Comment (UncleSamtoshi) [inline app/screens/send-bitcoin-screen/payment-details/onchain.ts:26]: Whats your thinking behind this?

**Review by UncleSamtoshi — CHANGES_REQUESTED** [review]

**Review by UncleSamtoshi — CHANGES_REQUESTED** [review]

File: app/components/text-currency/text-currency.tsx (hunk 1 above)
Comment (nicolasburtey) [reply app/components/text-currency/text-currency.tsx:33]: Yeah still in balance header
Plan to removed it when I'm doing a PR for it next

File: app/screens/send-bitcoin-screen/payment-details/onchain.ts (hunk 2 above)
Comment (nicolasburtey) [reply app/screens/send-bitcoin-screen/payment-details/onchain.ts:26]: Simplifying the Params passed via route. It makes mocking/testing screen easier/ less boilerplat-y. 

But the question here is: is it the same function as the one returned by the hook or do you wrap it somehow? Didn't check into details?

Comment (UncleSamtoshi) [reply app/screens/send-bitcoin-screen/payment-details/onchain.ts:26]: It is the same function from the hook, but it is important that the function is part of the object as it is used for the conversion to the settlement amount and it ensures that we can freeze the price at a particular point which is important when the user has confirmed a settlment amount and validation has been performed on that settlement amount.

File: app/components/text-currency/text-currency.tsx (hunk 1 above)
Comment (UncleSamtoshi) [reply app/components/text-currency/text-currency.tsx:33]: Sounds good!

File: app/screens/send-bitcoin-screen/payment-details/onchain.ts (hunk 2 above)
Comment (nicolasburtey) [reply app/screens/send-bitcoin-screen/payment-details/onchain.ts:26]: ok got it.

**Review by UncleSamtoshi — APPROVED** [review]

**Pull request merged**